src/python/pants/backend/go/target_type_rules.py

```diff
@@ -298,18 +304,16 @@ async def determine_main_pkg_for_go_binary(
     if len(relevant_pkg_targets) == 1:
         return GoBinaryMainPackage(relevant_pkg_targets[0].address)
 
-    wrapped_tgt = await Get(WrappedTarget, Address, addr)
-    alias = wrapped_tgt.target.alias
```

Comment:
I can add this back if we think it's worth it.

Reply:
Alternatively, should `FieldSet` store the target alias?

Reply:
Quite often one wants to fetch the `target` for the field set, is there any reason why the target is not kept in the field set, but just the address?

Reply:
I think primarily cache invalidation: a change to an irrelevant field should not invalidate memoization. Maybe this isn't as big of a deal as I thought, though.

Also, the idea was that the FieldSet should have already all the fields one needs -- no need to interact with the underlying Target.

--

But you know, I'm wondering if that API design is a bad one...Maybe it's too rigid! FieldSet does a bunch of things like no longer letting you see if `tgt.has_field()`. A simpler design could be nothing more than having `required_fields`, then storing the target as a field. You end up using the full Target API. 

That has a benefit of a much simpler API. You only interact with the conventional Target API.

But it does mean worse cache precision. cc @stuhood for thoughts